packages/grafana-ui/src/components/Combobox/MultiCombobox.tsx

```diff
@@ -340,42 +344,56 @@ export const MultiCombobox = <T extends string | number>(props: MultiComboboxPro
                   const index = virtualRow.index;
                   const item = options[index];
                   const itemProps = getItemProps({ item, index });
-                  const isSelected = isOptionSelected(item);
-                  const id = 'multicombobox-option-' + item.value.toString();
-                  const isAll = item.value === ALL_OPTION_VALUE;
+                  const isSelected = isComboboxOption(item) && isOptionSelected(item);
```

Comment:
Maybe doing a single call to isComboboxOption and reuse the value would make sense here?